include/boost/http_proto/message_base.hpp

```diff
@@ -95,7 +100,19 @@ class BOOST_SYMBOL_VISIBLE
     set_keep_alive(bool value);
 
 private:
-    char* set_prefix_impl(std::size_t);
+    struct prefix_holder {
+        message_base& mb_;
+        span<char> prefix_;
```

Comment:
why span instead of string_view?

Reply:
`span` is used because we're going to be copying into this, i.e. it's mutable.

`string_view` should really only be used for const views.